src/main/kotlin/no/nav/syfo/oppfolgingstilfelle/bit/Oppfolgingstilfellebit.kt

```diff
@@ -0,0 +1,142 @@
+    EGENMELDING,
+    FRAVAR_FOR_SYKMELDING,
+    PAPIRSYKMELDING,
+    ARBEID_GJENNOPPTATT,
+    KORRIGERT_ARBEIDSTID,
+    UKJENT_AKTIVITET,
+    UTDANNING,
+    FULLTID,
+    DELTID,
+    REDUSERT_ARBEIDSGIVERPERIODE,
+    REISETILSKUDD,
+    AVVENTENDE,
+}
+
+data class OppfolgingstilfelleBit(
+    val uuid: UUID,
+    val personIdentNumber: PersonIdentNumber,
+    val virksomhetsnummer: String? = null,
+    val createdAt: LocalDateTime,
+    val inntruffet: LocalDateTime,
+    val tagList: List<Tag>,
+    val ressursId: String,
+    val fom: LocalDateTime,
+    val tom: LocalDateTime,
+) {
+    companion object {
+        val TAG_PRIORITY: List<ListContainsPredicate<Tag>> = listOf(
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and Tag.ARBEID_GJENNOPPTATT,
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and Tag.KORRIGERT_ARBEIDSTID and Tag.BEHANDLINGSDAGER,
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and Tag.KORRIGERT_ARBEIDSTID and Tag.FULL_AKTIVITET,
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and Tag.KORRIGERT_ARBEIDSTID and (Tag.GRADERT_AKTIVITET or Tag.INGEN_AKTIVITET),
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and (Tag.PERMISJON or Tag.FERIE),
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and (Tag.EGENMELDING or Tag.PAPIRSYKMELDING or Tag.FRAVAR_FOR_SYKMELDING),
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and ListContainsPredicate.tagsSize(2),
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and Tag.BEHANDLINGSDAG,
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and Tag.BEHANDLINGSDAGER,
+            Tag.SYKMELDING and (Tag.SENDT or Tag.BEKREFTET) and Tag.PERIODE and Tag.BEHANDLINGSDAGER,
+            Tag.SYKMELDING and (Tag.SENDT or Tag.BEKREFTET) and Tag.PERIODE and Tag.FULL_AKTIVITET,
+            Tag.SYKMELDING and (Tag.SENDT or Tag.BEKREFTET) and Tag.PERIODE and (Tag.GRADERT_AKTIVITET or Tag.INGEN_AKTIVITET),
+            Tag.SYKMELDING and Tag.BEKREFTET and Tag.ANNET_FRAVAR,
+            Tag.SYKMELDING and Tag.SENDT and Tag.PERIODE and Tag.REISETILSKUDD and Tag.UKJENT_AKTIVITET,
+            Tag.SYKMELDING and Tag.NY and Tag.PERIODE and Tag.BEHANDLINGSDAGER,
+            Tag.SYKMELDING and Tag.NY and Tag.PERIODE and Tag.FULL_AKTIVITET,
+            Tag.SYKMELDING and Tag.NY and Tag.PERIODE and (Tag.GRADERT_AKTIVITET or Tag.INGEN_AKTIVITET),
+            Tag.SYKMELDING and Tag.NY and Tag.PERIODE and Tag.REISETILSKUDD and Tag.UKJENT_AKTIVITET,
+        )
+    }
+}
+
+fun List<OppfolgingstilfelleBit>.toOppfolgingstilfelleDagList(): List<OppfolgingstilfelleDag> {
+    require(this.isNotEmpty())
+
+    val firstFom = this.firstFom().toLocalDate()
+    val lastTom = this.lastTom().toLocalDate()
+
+    return (0..ChronoUnit.DAYS.between(firstFom, lastTom))
+        .map { dayInt ->
+            this.pickOppfolgingstilfelleDag(
+                dag = firstFom.plusDays(dayInt),
+            )
+        }
+}
+
+fun List<OppfolgingstilfelleBit>.firstFom(): LocalDateTime =
+    this.minByOrNull { it.fom }?.fom!!
```

Comment:
Synes kanskje 
`this.minByOrNull { it.fom }!!.fom`
er litt klarere (selv om semantiken antakelig blir den samme)

Reply:
Det blir jo hipp som happ det, så kan godt endre til varianten du foreslår.